### DISCUSSION TITLE
This is the ~4th time I upgrade DBeaver and my workspace is GONE

### DISCUSSION THREAD

**Discussion by RayNeverLearns** [dbeaver/dbeaver#15306] [Q&A]
I just upgraded to v21.3.3 from v21.3.2 and as soon as I open it, I am asked to create a new db connection. My entire workspace is GONE! All of my connections and all of my open query windows I had (over 15). What's worse is that this is at least the 4th time it's happened in the past 1.5-2 years that I've been using DBeaver and it only happens when I upgrade. 

%APPDATA%\Roaming\DBeaverData\workspace6\ is back to default (no connections, no scripts) and this is very frustrating. Is this a known issue and is there a way around that other than saving the entire workspace folder structure every single time before I upgrade? I now have to start ALL this from scratch all over again. Days and days of work.

**Comment by RayNeverLearns** [comment]
Nobody else experienced this?

**Reply by p91paul** [reply]
I for one have been continuously upgrading DBeaver from some years now and this never happened.

**Reply by RayNeverLearns** [reply]
I wish I could say the same or at least that this is the first time it happens to me. I had about 15 connections configured to different DBMS's from SQL Server to Oracle to DB2 and on different platforms from Windows to Unix to z/OS, all of which use different credentials that I may or may not have stored elsewhere. I'm sure you can imagine what a headache this is to recreate. Plus many scripts I've been working on for months that I never bothered to save since DBeaver always does by itself. I will need to save my entire workspace before upgrading from now on. But getting to where I was before the upgrade will take me weeks now, not days as I originally anticipated.
